### ISSUE TITLE
Composer 2.0 package-name // Uppercase name will error

### ISSUE THREAD

**Issue by sebastian-marinescu** [modmore/Gitify#252]
### Summary
Composer 2.0 wants package-names to be lowercase

### Step to reproduce
`composer install` on a server

### Observed behavior
Throws a deprecation warning: 

> Deprecation warning: Your package name modmore/Gitify is invalid, it should not contain uppercase characters. We suggest using modmore/gitify instead. Make sure you fix this as Composer 2.0 will error.

### Environment
Composer version 1.8.1 2019-01-29
Gitify version 0.12.0

**Comment by Mark-H** [comment]
Also:

````
Deprecation warning: require.kbjr/Git.php is invalid, it should not contain uppercase characters. Please use kbjr/git.php instead. Make sure you fix this as Composer 2.0 will error.
````

**Comment by sebastian-marinescu** [comment]
Yes, already issued: 
https://github.com/kbjr/Git.php/issues/63

Maybe I'll test a PR there first and see the implications :smirk:

**Comment by sebastian-marinescu** [comment]
@Mark-H I've just asked myself: "What is Gitify even using `Git.php` even for?"

Doing some detective work I see it's used in the function `getGitRepository()`: 
https://github.com/modmore/Gitify/blob/master/src/Gitify.php#L104-L123
which in return is not used anywhere. 

Also the system setting `gitify.git_path` is not existent in the Gitify, nor in the GitifyWatch repositories. 

It is mentioned in the changelog and you introduced it back in 2015: 
https://github.com/modmore/Gitify/commit/fb05c2d7e8f9ed685e7c8b9a744d407bfdd11b22#diff-4ac32a78649ca5bdd8e0ba38b7006a1e

So, is this some feature prepared for something in the future? 
Do you know what you initially wanted to do here? :)

**Comment by sebastian-marinescu** [comment]
Ok, investigating a little further, I see it is being used by GitifyWatch here: 
https://github.com/modmore/GitifyWatch/blob/master/core/components/gitifywatch/model/gitifywatch/gitifywatch.class.php#L60-L78

The GitifyWatch process loads the Gitify-instance and then uses `getGitRepository()` to get the repository-object from `\Git`. 

If I understand now correctly - isn't this function `getGitRepository()` _and_ the dependency to "Git.php" better off in the GitifyWatch-code?

**Comment by Mark-H** [comment]
I guess at some point the idea was to add more git stuff directly into Gitify, or to have other things interact with it through the gitify class, making that the single place to get the repo object.

**Comment by sebastian-marinescu** [comment]
So this issue got resolved successfully in the [Git.php](https://github.com/kbjr/Git.php/issues/63) project. 

There is now another issue about that #323 and this PR #324 will fix both #323/#252.
